src/medCore/database/medDatabaseExporter.cpp

```diff
@@ -59,7 +73,18 @@ void medDatabaseExporter::internalRun(void)
     }
 
     dtkAbstractDataWriter * dataWriter = medAbstractDataFactory::instance()->writer(d->writer);
-    dataWriter->setData(d->data);
+    if(!d->saveMultipleData)
+    {
+        dataWriter->setData(d->data);
+    }
+    else
+    {
+        medAbstractDataWriter* medDataWriter = dynamic_cast<medAbstractDataWriter*>(dataWriter);
+        if(medDataWriter)
```

Comment:
My initial comment was just to point out that if one wants to test the success of the cast then a try-catch won't work. On the other hand, I am not for putting these kinds of tests in the production code because they are a form of unit test. Like @LoicCadour said, it would be a dev error and so should cause a crash, but the tests for this kind of crash should normally be in the form an assert without an ```if``` block: 
```
medAbstractDataWriter* medDataWriter = dynamic_cast<medAbstractDataWriter*>(dataWriter);
Q_ASSERT(medDataWriter != NULL)
blah blah...
```